### PR TITLE
Fix robots noindex signals

### PR DESCRIPTION
**Pull request by nickvergessen**
cc @tkirl

### CODE REVIEW

**Review by kesselb — APPROVED** [review]

**Comment by kesselb** [comment]
Looks good to me ;) Do we still need the `robots.txt`?

**Comment by nickvergessen** [comment]
Well better save then sorry we add all possible options.

**Review by rullzer — APPROVED** [review]

**Comment by rullzer** [comment]
CI says no

**Comment by nickvergessen** [comment]
Fixed

**Review by blizzz — APPROVED** [review]

File: lib/public/AppFramework/Http/TemplateResponse.php [hunk 1]
```diff
@@ -157,6 +157,7 @@ public function render() {
 		// \OCP\Template needs an empty string instead of 'blank' for an unwrapped response
 		$renderAs = $this->renderAs === 'blank' ? '' : $this->renderAs;
 
+		\OCP\Util::addHeader('meta', ['name' => 'robots', 'content' => 'noindex']);
```
Comment (blizzz) [inline lib/public/AppFramework/Http/TemplateResponse.php:160]:
```suggestion
		\OCP\Util::addHeader('meta', ['name' => 'robots', 'content' => 'noindex, nofollow']);
```

(as above)

File: lib/public/AppFramework/Http/Response.php [hunk 2]
```diff
@@ -238,6 +238,7 @@ public function getHeaders() {
 
 		$this->headers['Content-Security-Policy'] = $this->getContentSecurityPolicy()->buildPolicy();
 		$this->headers['Feature-Policy'] = $this->getFeaturePolicy()->buildPolicy();
+		$this->headers['X-Robots-Tag'] = 'noindex';
```
Comment (blizzz) [inline lib/public/AppFramework/Http/Response.php:241]:
```suggestion
		$this->headers['X-Robots-Tag'] = 'noindex, nofollow';
```

Will also avoid following links uselessly, cf https://developers.google.com/search/reference/robots_meta_tag The original robots.txt would have prevented to enter the site generally.

File: tests/lib/AppFramework/Controller/ControllerTest.php [hunk 3]
```diff
@@ -114,6 +114,7 @@ public function testFormatDataResponseJSON() {
 			'Content-Type' => 'application/json; charset=utf-8',
 			'Content-Security-Policy' => "default-src 'none';base-uri 'none';manifest-src 'self'",
 			'Feature-Policy' => "autoplay 'none';camera 'none';fullscreen 'none';geolocation 'none';microphone 'none';payment 'none'",
+			'X-Robots-Tag' => 'noindex',
```
Comment (blizzz) [inline tests/lib/AppFramework/Controller/ControllerTest.php:117]:
```suggestion
			'X-Robots-Tag' => 'noindex, nofollow',
```

File: tests/lib/AppFramework/Http/DataResponseTest.php [hunk 4]
```diff
@@ -67,6 +67,7 @@ public function testConstructorAllowsToSetHeaders() {
 			'Cache-Control' => 'no-cache, no-store, must-revalidate',
 			'Content-Security-Policy' => "default-src 'none';base-uri 'none';manifest-src 'self'",
 			'Feature-Policy' => "autoplay 'none';camera 'none';fullscreen 'none';geolocation 'none';microphone 'none';payment 'none'",
+			'X-Robots-Tag' => 'noindex',
```
Comment (blizzz) [inline tests/lib/AppFramework/Http/DataResponseTest.php:70]:
```suggestion
			'X-Robots-Tag' => 'noindex, nofollow',
```

File: tests/lib/AppFramework/Http/ResponseTest.php [hunk 5]
```diff
@@ -51,7 +51,8 @@ public function testSetHeaders() {
 		$expected = [
 			'Last-Modified' => 1,
 			'ETag' => 3,
-			'Something-Else' => 'hi'
+			'Something-Else' => 'hi',
+			'X-Robots-Tag' => 'noindex',
```
Comment (blizzz) [inline tests/lib/AppFramework/Http/ResponseTest.php:55]:
```suggestion
			'X-Robots-Tag' => 'noindex, nofollow',
```

**Comment by rullzer** [comment]
@nickvergessen mind to checkout the comments by @blizzz

**Comment by nickvergessen** [comment]
Done, went with `none` which as per https://developers.google.com/search/reference/robots_meta_tag is an equivalent for `noindex, nofollow`.
ALso while replacing I noticed we try to set the header in the htaccess as well and also have a setup check for this. So I guess it's already working all the time as expected. But this here should help nginx users by default

**Pull request merged**